### PR TITLE
Automatically check license compatibility

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

Use [`cargo-about`](https://github.com/EmbarkStudios/cargo-about) to automatically check license compatibility.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Checked the number of crates that `cargo-about` gathered against the number of crates in `Cargo.lock`. I also checked if the workflow fails if licenses are left out.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Comment by grtlr** [comment]
We still need to decide which licenses we want to accept. The current set of licenses can be found in `about.toml`. @thibault-martinez what do you think?

**Comment by thibault-martinez** [comment]
Is that all the licenses we are currently having in our deps ? I would just start with that and add more in the future as they come up and if they are compatible.

**Comment by grtlr** [comment]
Yes, that's what I'm going with for now. I don't know how these licenses interact though—especially `ISC` and `MPL`.

File: about.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "Apache-2.0",
+    "BSD-3-Clause",
+    "CC0-1.0",
+    "MIT",
+    "MPL-2.0",
+]
+# For now we include these so that the number of gathered crates matches the
+# number of crates in `Cargo.lock`.
+ignore-build-dependencies = false
+ignore-dev-dependencies = false
```
Comment (thibault-martinez) [inline about.toml]: Can you maybe put this file in the `.github/workflows` folder please ?

File: .github/workflows/about.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# This is the configuration file that is used by `cargo-about` to check license
```
Comment (thibault-martinez) [inline .github/workflows/about.toml]: As a note for the future, we use a format with 120 columns, including comments. This is not code per se so you don't necessarily have to change here but keep in mind for later ;)

Comment (grtlr) [reply .github/workflows/about.toml]: Good to know! I have changed the formatting.

**Review by thibault-martinez — APPROVED** [review]

**Review by pvdrz — DISMISSED** [review]

**Review by Adam-Gleave — DISMISSED** [review]

File: .github/workflows/licenses.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - stable
+      - coordicide
+  pull_request:
+    branches:
+      - dev
+      - stable
+      - coordicide
+
+jobs:
+  format:
```
Comment (thibault-martinez) [inline .github/workflows/licenses.yml]:
```suggestion
  licenses:
```
Probably copy-pasted 😬

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: .github/workflows/licenses.yml (hunk 3 above)
Comment (grtlr) [reply .github/workflows/licenses.yml]: you bet

**Review by thibault-martinez — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Pull request merged**